public/pages/CreateDetector/components/ConfigureAlerts/containers/ConfigureAlerts.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { EuiButton, EuiSpacer, EuiTitle } from '@elastic/eui';
+import { CREATE_DETECTOR_STEPS } from '../../../utils/constants';
+import {
+  EMPTY_DEFAULT_ALERT_CONDITION,
+  MAX_ALERT_CONDITIONS,
+  MIN_ALERT_CONDITIONS,
+} from '../utils/constants';
+import AlertConditionPanel from '../components/AlertCondition';
+import { Detector } from '../../../../../../models/interfaces';
+
+interface ConfigureAlertsProps extends RouteComponentProps {
+  changeDetector: (detector: Detector) => void;
+  detector: Detector;
+  isEdit: boolean;
+}
+
+interface ConfigureAlertsState {
+  loading: boolean;
+  notificationChannels: string[];
+  ruleTypes: string[];
+}
+
+export default class ConfigureAlerts extends Component<ConfigureAlertsProps, ConfigureAlertsState> {
+  constructor(props: ConfigureAlertsProps) {
+    super(props);
+    this.state = {
+      loading: false,
+      notificationChannels: [],
+      ruleTypes: [],
+    };
+  }
+
+  componentDidMount = async () => {
+    const {
+      detector: { alert_conditions },
+    } = this.props;
+    this.getNotificationChannels();
+    if (alert_conditions.length < MIN_ALERT_CONDITIONS) {
+      this.addCondition();
+    }
+  };
+
+  getNotificationChannels = async () => {
+    this.setState({ loading: true });
+    this.setState({ loading: false });
+  };
```

Comment:
Curious about this part.  Why async, and why setting state twice in a row?

Reply:
The async was added for when we will make rest calls to fetch the data, but yeah the duplicate line is an error, let me fix it.

Reply:
sounds good.

Reply:
The lines differ in the state value actually, added TODO comment to make the difference apparent